packages/astro/src/@types/astro.ts

```diff
@@ -2408,6 +2408,126 @@ export interface AstroUserConfig {
 		 * For a complete overview and the full API reference, see [the Content Layer API RFC](https://github.com/withastro/roadmap/blob/content-layer/proposals/0047-content-layer.md) and [share your feedback](https://github.com/withastro/roadmap/pull/982).
 		 */
 		contentLayer?: boolean;
+
+		/**
+		 * @docs
+		 * @name experimental.tsconfig
+		 * @type {object}
+		 * @default `undefined`
+		 * @version 4.15.0
+		 * @description
+		 *
+		 * Enables a streamlined way of configuring TypeScript in your project. Enjoy an improved DX when setting up Astro for Intellisense, type-checking, and more.
+		 *
+		 * This option removes the need for a separate `src/env.d.ts` file to provide Vite's client types to your project. Instead, `/.astro/tsconfig.json` will be autogenerated with linked types for you. Notably, this fixes an existing issue with `astro check` false positives by excluding `outDir` from your checked files by default.
+		 *
+		 * To enable this option in an existing project, you will need to update your `tsconfig.json` to reference the new file. You may also need to update your `astro.config.mjs` as some items have now moved to Astro configuration options.
+		 *
+		 * Your `tsconfig.json` file must now extend the new `/.astro/` file:
+		 *
+		 * ```json title="tsconfig.json" ins="'./.astro/tsconfig.json'"
+		 * {
+		 * 	 extends: ['astro/tsconfigs/base', './.astro/tsconfig.json']
+		 * }
+		 * ```
+		 *
+		 * If you have `include` or `exclude` inside your root `tsconfig.json`, move them to your Astro config:
+		 *
+		 * ```json title="tsconfig.json" del={2-3}
+		 * {
+		 *   include: ['src/**/*'],
+		 *   exclude: ['node_modules']
+		 * }
+		 * ```
+		 *
+		 * ```js title="astro.config.*" ins={6-7}
+		 * import { defineConfig } from 'astro/config'
+		 *
+		 * export default defineConfig({
+		 *   experimental: {
+		 *    tsconfig: {
+		 *      include: ['src/**/*'],
+		 *      exclude: ['node_modules']
+		 *    }
+		 *   }
+		 * })
+		 * ```
```

Comment:
```suggestion
		 * ```
		 * 
		 * For a complete overview, and to give feedback on this experimental API,
     * see the [Feature RFC](https://github.com/withastro/roadmap/blob/actions/proposals/00xx-feature.md).
```
Is there a link it makes sense to put here?

Reply:
Unfortunately it's not part of any RFC so nothing to link